### DISCUSSION TITLE
is it possible to pair a Sonoff SNZB-04 via a router instead of the coordinator?

### DISCUSSION THREAD

**Discussion by paopre** [Koenkk/zigbee2mqtt#10989] [Devices]
Hello,

does anyone have experience with Sonoff SNZB-04 door sensor?
I need to know if it can be included (paired) in the network via a router.
The following thread says it’s not possible with a 03 model:

https://github.com/Koenkk/zigbee2mqtt/issues/6979

But what about 04?

Thanks!

**Comment by JackTalisker** [comment]
I have some problems with 3 SNZB-04 sensors. 
I paired them near the coordinator then I positioned them. 
As you can see in the attached image, they have lost the connection with the coordinator without establishing a new route (there is a CC2531 router and an ikea repeater very colse). 
I tried to pair them again near the repeater but no way
![z2m_map](https://user-images.githubusercontent.com/68334069/150750340-d912046d-2a46-4afb-8ab6-9dd400627151.PNG)
.

**Reply by paopre** [reply]
Thanks for the infos

1) Did you wait some hours (or even a day) in order to let the SNZB-04 sensors wake up and estabilish a new route?
2) what about the other door sensors? Do they work? Which model do you use?

**Reply by JackTalisker** [reply]
I've been struggling for 3 days in order to make those sensor working and not sure how long it will last. Sometimes they give an unavailable state (in HA) instead of close/open.
The weird thing is that if one sensor gives the unavailable state in HA (when tickled), all the other SNZB-04 in HA gives unavailable state at the same time.
I (re)paired them many times and still experimenting. My coordinator is a tasmotized Sonoff.

**Reply by sjorge** [reply]
From my experience, they only chose a new parent router if the battery is removed and inserted a few seconds later. (No need to pair again)

They do however sleep most of the time and after an hour or so the map won't show a line, but they do keep working for me. (Until the battery dies or coordinator is down when they send a message)

**Reply by paopre** [reply]
> I've been struggling for 3 days in order to make those sensor working and not sure how long it will last. Sometimes they give an unavailable state (in HA) instead of close/open. The weird thing is that if one sensor gives the unavailable state in HA (when tickled), all the other SNZB-04 in HA gives unavailable state at the same time. I (re)paired them many times and still experimenting. My coordinator is a tasmotized Sonoff.

I've had the same problems. In order to prevent the "all sensors unavailable" problem, I use a snapshot of my HA installation, so I don't need to re-pair the sensors. In addition: I figured that they are totally unpredictable/unmanageable in routing: therefore I think that the only solution is to link them ONLY to coordinators and use multiple coordinators (like the tasmotized sonoff you use)

BTW: from the attached picture of your network, it seems you are italian, like me. Could you find xiaomi sensors in Italy at a good/decent price? They seem very hard to find. I would like to test them as well.

**Reply by JackTalisker** [reply]
> From my experience, they only chose a new parent router if the battery is removed and inserted a few seconds later. (No need to pair again)
> 
> They do however sleep most of the time and after an hour or so the map won't show a line, but they do keep working for me. (Until the battery dies or coordinator is down when they send a message)

thank you sjorge, I'll try the "battery removed" trick

**Reply by JackTalisker** [reply]
> > I've been struggling for 3 days in order to make those sensor working and not sure how long it will last. Sometimes they give an unavailable state (in HA) instead of close/open. The weird thing is that if one sensor gives the unavailable state in HA (when tickled), all the other SNZB-04 in HA gives unavailable state at the same time. I (re)paired them many times and still experimenting. My coordinator is a tasmotized Sonoff.
> 
> I've had the same problems. In order to prevent the "all sensors unavailable" problem, I use a snapshot of my HA installation, so I don't need to re-pair the sensors. In addition: I figured that they are totally unpredictable/unmanageable in routing: therefore I think that the only solution is to link them ONLY to coordinators and use multiple coordinators (like the tasmotized sonoff you use)

I've been trying this new z2m installation for about a week mostly out of curiosity. Before I had ZHA installed and except for very few incompatible devices, it was working very well and reliably. I hadn't had any issues with snzb-04 sensors; the only sensor which drives me crazy either with zha or with z2m is the ikea tradfri motion sensor which seems to go sleeping after few hours and needs to be awakened touching one of his buttons from time to time.
> 
> BTW: from the attached picture of your network, it seems you are italian, like me. Could you find xiaomi sensors in Italy at a good/decent price? They seem very hard to find. I would like to test them as well.

Yes I'm italian. I own two Xiaomi Aqara temp/hum sensors, two Aqara water leak sensors and one Aqara motion/luminance sensor. They are all good and reilable sensors.
I bought them a couple of years ago in china (Ali) when the prices were still decent.

**Comment by FreddyG88** [comment]
Hello together,

does anyone have an update in this topic? I have the same issue, using a Sonoff 3.0 USB dongle with launchpad router FW and the same dongle with Coordinator FW as coordinator for Zigbee2MQTT. All SNZB-04 sensors which are close to the coordinator are working well, but now I installed 3 further SNZB-04 sensors 2 floors above the coordinator which means they need to take the router. I have some develco smoke detectors on the same floor which are working fine via the router. The door sensors can be paired via router and work for several minutes afterwards, but after a while they do not update the status anymore. I observed that other sensors which are close enough to corrdinator, but much closer to the router prefer the direct connection to ther coordinator, too.
